bqskit/compiler/registry.py

```diff
@@ -34,10 +38,13 @@ def register_workflow(
             be executed if the MachineModel in a call to `compile` matches
             `key`. If `key` is already registered, a warning will be logged.
 
-        optimization_level ptional[int): The optimization level with which
+        optimization_level (Optional[int]): The optimization level with which
             to register the workflow. If no level is provided, the Workflow
             will be registered as level 1.
 
+        target_type (str): Register a workflow for targets of this type. Must
+            be 'circuit', 'unitary', 'stateprep', or 'statemap'.
```

Comment:
Should this default to circuit?

Reply:
I'm not opposed to this. I think if this happens, there should be a check to see if a `'circuit'` (or any other) flow has been registered so that a warning can be made just in case.